ic-agent/src/agent/mod.rs

```diff
@@ -330,12 +330,21 @@ impl Agent {
     }
 
     fn get_expiry_date(&self) -> u64 {
-        // TODO(hansl): evaluate if we need this on the agent side (my hunch is we don't).
-        let permitted_drift = Duration::from_secs(60);
-        self.ingress_expiry
-            .as_nanos()
-            .saturating_add(OffsetDateTime::now_utc().unix_timestamp_nanos() as u128)
-            .saturating_sub(permitted_drift.as_nanos()) as u64
+        let expiry_raw = OffsetDateTime::now_utc() + self.ingress_expiry;
+        let rounded = if self.ingress_expiry.as_secs() > 60 {
+            let mut rounded_minute = expiry_raw
+                .replace_second(0)
+                .unwrap()
+                .replace_nanosecond(0)
+                .unwrap();
+            if expiry_raw.second() >= 30 {
+                rounded_minute += Duration::from_secs(60);
+            }
+            rounded_minute
+        } else {
+            expiry_raw.replace_nanosecond(0).unwrap()
+        };
```

Comment:
It seems like we're unconditionally doing `.replace_nanosecond(0)`.  Could we do it in one place?